plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -270,6 +280,70 @@ private void testAddNestedColumnWithColumnMappingMode(String columnMappingMode)
                 .containsPattern("(delta\\.columnMapping\\.physicalName.*?){11}");
     }
 
+    @Test // regression test for https://github.com/trinodb/trino/issues/24121
+    void testPartitionValuesParsedCheckpoint()
+            throws Exception
+    {
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.ID);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NAME);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NONE);
+    }
+
+    private void testPartitionValuesParsedCheckpoint(ColumnMappingMode columnMappingMode)
```

Comment:
Should we have also product test in `TestDeltaLakeColumnMappingMode` to check reading/writing checkpoints by trino/delta